pulp_smash/tests/rpm/api_v2/test_updateinfo.py

```diff
@@ -331,52 +356,58 @@ class PkglistsTestCase(unittest.TestCase):
     def test_all(self):
         """Sync a repo whose updateinfo file has multiple pkglist sections.
 
-        Do the following:
+        Specifically, do the following:
 
-        1. Create and sync a repository with an importer and distributor.
-           Ensure the importer's feed is set to
+        1. Create, sync and publish an RPM repository whose feed is set to
            :data:`pulp_smash.constants.RPM_PKGLISTS_UPDATEINFO_FEED_URL`.
-        2. Publish the repository, and fetch and parse its updateinfo file.
-        3. Verify the updateinfo contains the correct number of ``<pkglists>``
-           sections, with the correct contents in each.
+        2. Fetch and parse the published repository's ``updateinfo.xml`` file.
+
+        Verify that the ``updateinfo.xml`` file has three packages whose
+        ``<filename>`` elements have the following text:
+
+        * penguin-0.9.1-1.noarch.rpm
+        * shark-0.1-1.noarch.rpm
+        * walrus-5.21-1.noarch.rpm
+
+        Note that Pulp is free to change the structure of a source repository
+        at will. For example, the source repository has three ``<collection>``
+        elements, the published repository can have one, two or three
+        ``<collection>`` elements. Assertions are not made about these details.
         """
         cfg = config.get_config()
         if selectors.bug_is_untestable(2227, cfg.version):
             self.skipTest('https://pulp.plan.io/issues/2277')
 
-        # Create and sync a repository.
+        # Create, sync and publish a repository.
         client = api.Client(cfg, api.json_handler)
         body = gen_repo()
         body['importer_config']['feed'] = RPM_PKGLISTS_UPDATEINFO_FEED_URL
         body['distributors'] = [gen_distributor()]
         repo = client.post(REPOSITORY_PATH, body)
-        repo = client.get(repo['_href'], params={'details': True})
         self.addCleanup(client.delete, repo['_href'])
+        repo = client.get(repo['_href'], params={'details': True})
```

Comment:
I'm moving this line of code down so that `addCleanup()` executes as soon as possible.